lms/djangoapps/course_blocks/transformers/start_date.py

```diff
@@ -43,6 +43,24 @@ def name(cls):
         """
         return "start_date"
 
+    @classmethod
+    def _check_has_scheduled_content(cls, block_structure, scheduled_content_condition):
+        '''
+        Returns a block structure where the root course block has been
+        updated to include a has_scheduled_content field (True if the course
+        has any blocks with release dates in the future, False otherwise).
+        '''
+        has_scheduled_content = False
+        for block_key in block_structure.topological_traversal():
+            if scheduled_content_condition(block_key):
+                has_scheduled_content = True
+                break
+
+        if has_scheduled_content:
+            block_structure.override_xblock_field(
+                block_structure.root_block_usage_key, 'has_scheduled_content', has_scheduled_content
+            )
```

Comment:
Nit: just put the override xblock field right after the if and get rid of the has_scheduled_content variable. It doesn't really have any use in this function.

Reply:
Ooh yeah idk what the if on 59 is about. I think this is copy pasta from an earlier attempt. We want to set has_scheduled_content whether it's true or false...